### PR TITLE
Adds basic crate structure

### PR DESCRIPTION
**Pull request by tarikeshaq**
Adds basic project structure

### CODE REVIEW

File: experiments/Cargo.toml [hunk 1]
```diff
@@ -15,3 +15,9 @@ crate-type = ["lib"]
 
 [dependencies]
 anyhow = "1"
+serde = "1"
+serde_derive = "1"
+log = "0.4"
+viaduct = { git = "https://github.com/mozilla/application-services" }
```
Comment (vladikoff) [inline experiments/Cargo.toml:21]: can we add a GIT SHA here? Lock to a specific release maybe?

File: experiments/src/buckets.rs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! This might be where the bucketing logic can go
```
Comment (vladikoff) [inline experiments/src/buckets.rs:5]: @tarikeshaq while porting this to Rust, does it make sense to call this file `evaluator`, to mimic what https://github.com/mozilla-mobile/android-components/blob/master/components/service/experiments/src/main/java/mozilla/components/service/experiments/ExperimentEvaluator.kt does (bucketing and more)?

File: experiments/src/matcher.rs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+//! This module defines all the information needed to match a user with an experiment
+//! this module should also include a `match` function of some sort that does the matching
+//! it has two main types, the `matcher` retrieved from the server, and the `AppContext`
+//! from the client
+//! Note: This could be where the logic to evaluate the filter_expressions lies
+use serde_derive::*;
+
+#[derive(Deserialize, Serialize, Debug, Clone, Default)]
+pub struct Matcher {
+    pub app_id: Option<String>,
```
Comment (vladikoff) [inline experiments/src/matcher.rs:14]: @pdehaan do you have a list of fields that Firefox Desktop implementation provides for Experiment context? Are we missing anything?

File: experiments/src/metrics_manager.rs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (vladikoff) [inline experiments/src/metrics_manager.rs:1]: We can remove this file for now I think

File: experiments/src/buckets.rs (hunk 2 above)
Comment (travis79) [reply experiments/src/buckets.rs]: I second this, if the plan is to have the bucketing and matching evaluation in this module then 'evaluator' makes more sense to me.

File: experiments/src/lib.rs [hunk 5]
```diff
@@ -2,9 +2,17 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
+mod buckets;
```
Comment (travis79) [inline experiments/src/lib.rs:5]: This should be `evaluator` now, right?

File: experiments/src/matcher.rs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    pub app_display_version: Option<String>,
+    pub app_min_version: Option<String>,
+    pub app_max_version: Option<String>,
+    pub locale_language: Option<String>,
+    pub locale_country: Option<String>,
```
Comment (travis79) [inline experiments/src/matcher.rs:19]: service/experiments splits locale into country and language but Glean records this as the combined format (en-US, en-GB, sp-MX, de-DE, etc..)  I propose combining these into `locale` and going with the same format we use in telemetry to make analysis easier for the data science folks.

File: experiments/src/lib.rs (hunk 5 above)
Comment (tarikeshaq) [reply experiments/src/lib.rs]: Yup! Good catch 🤦

File: experiments/src/matcher.rs (hunk 3 above)
Comment (travis79) [reply experiments/src/matcher.rs:14]: I believe Desktop is using JEXL for their targeting/matching, and the only docs I found on that [are here](https://firefox-source-docs.mozilla.org/browser/components/newtab/content-src/asrouter/docs/targeting-guide.html?highlight=jexl%20targeting)

**Pull request merged**